File: 2D_amber/Assets/Scripts/CSharp/Behaviour/MovementBehaviour.cs
using UnityEngine;
using System.Collections;

public class MovementBehaviour : StateMachineBehaviour {
	void OnStateEnter(Animator anim, AnimatorStateInfo info, int id){
    	anim.ResetTrigger("Skill") ;
    	anim.SetBool("Shot", false) ;
    	anim.GetComponentInParent<Controller>().lockMove = false ;
  	}
  	
  	void OnStateUpdate(Animator anim, AnimatorStateInfo info, int id){
    	Game.sav.RegenSP(Time.deltaTime) ;
  	}
}

File: 2D_amber/Assets/Scripts/JS/Behaviour/MovementBehaviour.js
#pragma strict

class MovementBehaviour extends StateMachineBehaviour{
  function OnStateEnter(anim:Animator, info:AnimatorStateInfo, id:int){
    anim.ResetTrigger("Skill") ;
    anim.SetBool("Shot", false) ;
    anim.GetComponentInParent.<Controller>().lockMove = false ;
  }
  
  function OnStateUpdate(anim:Animator, info:AnimatorStateInfo, id:int){
    Game.sav.RegenSP(Time.deltaTime) ; 
  }
}
